src/platform/tls_schannel.c

```diff
@@ -2505,11 +2505,19 @@ CxPlatTlsWriteDataToSchannel(
                         Result |= CXPLAT_TLS_RESULT_ERROR;
                         break;
                     }
-                    State->ReadKey = QUIC_PACKET_KEY_1_RTT;
-                    QuicTraceLogConnInfo(
-                        SchannelRead1RttStart,
-                        TlsContext->Connection,
-                        "Reading 1-RTT data starts now");
+                    //
```

Comment:
I'm not sure about these TLS layer changes... Why can't this be abstracted away at the QUIC layer, like we did do before?

Reply:
Not sure where would be the best place for that, I thought it better to reflect the condition in the ReadKey value itself and not add additional conditions on multiple places. Elsewhere.

Do you have any suggestions?

Reply:
We should defer packet processing at the QUIC layer until we're ready.

Reply:
I see, so you think it would be better somewhere like `QuicConnGetKeyOrDeferDatagram` and keep the TLS layer intact?

Reply:
Yeah. I think so.